### PR TITLE
hep: use enum for energy ranges

### PR DESCRIPTION
**Pull request by michamos**
Based on @ksachs's input.

### CODE REVIEW

File: inspire_schemas/records/hep.yml [hunk 1]
```diff
@@ -684,12 +684,21 @@ properties:
...
-            type: integer
+            enum:
+            - 0-3 GeV
+            - 3-10 GeV
+            - 10-30 GeV
+            - 30-100 GeV
+            - 100-300 GeV
+            - 300-1000 GeV
+            - 1-10 TeV
+            - '> 10 TeV'
```
Comment (michamos) [inline inspire_schemas/records/hep.yml]: this is quoted because `>` has special meaning in yaml.

Comment (david-caro) [inline inspire_schemas/records/hep.yml]: \o/ hurray yaml extensive set of hidden/quirk features!

Comment (david-caro) [inline inspire_schemas/records/hep.yml]: Are the ranges inclusive to the left?
ex. 3-10 == (>=3 && <10) ?

Comment (david-caro) [inline inspire_schemas/records/hep.yml]: Maybe the comment should say something about that (I would be confused, I'm not a curator though...

Comment (ksachs) [inline inspire_schemas/records/hep.yml]: it's inclusive on the right. I.e. 695__e:6 ( 300-1000 GeV) includes 1 TeV

Comment (michamos) [inline inspire_schemas/records/hep.yml]: I documented it in an amended commit. Could someone approve this?

**Pull request merged**